### PR TITLE
Add a stress test with 100 CRDs (#804)

### PR DESCRIPTION
**Pull request by victorpras**
We've found OOMKills in ACM 1.16.0 on Autopilot clusters with 100+ CRDs. This commit adds a stress test to make sure Config Sync can still work with those CRDs.

`TestStress100CRDs` is needed to cherry-pick nomos CLI client side throttling fix (https://github.com/GoogleContainerTools/kpt-config-sync/pull/853)

### CODE REVIEW

File: e2e/testcases/stress_test.go [hunk 1]
```diff
@@ -261,6 +265,91 @@ func TestStressLargeRequest(t *testing.T) {
 	}
 }
 
+func TestStress100CRDs(t *testing.T) {
```
Comment (nan-yu) [inline e2e/testcases/stress_test.go:268]: The indentation needs to be fixed.

Comment (victorpras) [reply e2e/testcases/stress_test.go:268]: what's wrong with the indentation? Running `make fmt-go` doesn't yield any issue

Comment (nan-yu) [reply e2e/testcases/stress_test.go:268]: image

Comment (victorpras) [reply e2e/testcases/stress_test.go:268]: I'm not seeing the same indentation in my side screen/8U8jBpSxF4wvYFy, could this be due different github settings https://stackoverflow.com/questions/70774875/problems-with-indentations-on-github

Comment (nan-yu) [reply e2e/testcases/stress_test.go:268]: Restart the browser, and it looks good now.

**Review by nan-yu — APPROVED** [review]
/lgtm

**Pull request merged**